perma_web/perma/tasks.py

```diff
@@ -2204,21 +2204,65 @@ def upload_link_to_internet_archive(link_guid, attempts=0, timeouts=0):
         logger.info(f"Queued Link {link_guid} no longer eligible for upload.")
         return
 
-    # Make sure we've already created the IA item for this link
+    # Get or create the appropriate InternetArchiveItem object for this link
+    date_string = link.creation_timestamp.strftime('%Y-%m-%d')
     identifier = InternetArchiveItem.DAILY_IDENTIFIER.format(
         prefix=settings.INTERNET_ARCHIVE_DAILY_IDENTIFIER_PREFIX,
-        date_string=link.creation_timestamp.strftime('%Y-%m-%d')
+        date_string=date_string
     )
-    try:
-        perma_item = InternetArchiveItem.objects.get(identifier=identifier)
-    except InternetArchiveItem.DoesNotExist:
-        msg = f"Internet Archive Item {identifier} does not exist: not uploading {link_guid}."
-        if settings.INTERNET_ARCHIVE_EXCEPTION_IF_NO_ITEM:
-            logger.error(msg)
+    start = InternetArchiveItem.datetime(f"{date_string} 00:00:00")
+    end = start + timedelta(days=1)
+    perma_item, _created = InternetArchiveItem.objects.get_or_create(
+        identifier=identifier,
+        span=(start, end),
+    )
+
+    # Check the Internet Archive-related status of this link
+    perma_file = InternetArchiveFile.objects.filter(
+        item_id=identifier,
+        link_id=link_guid
+    ).first()
+    if perma_file:
+        if perma_file.status == 'confirmed_present':
```

Comment:
When you use an `enum` for things like this, do you feel comfortable omitting the `else` at the end of switch-style blocks? I find myself still including it... maybe coding against the possibility that in the future, another legal value is added, but we forget to account for it.

Reply:
Thanks for suggesting the enum. I am used to relying on Django's validation of choices to reject invalid values.... but since we only see that if calling `full_clean`, which we don't routinely do, much better idea!

Reply:
In general I avoid "just in case" code because I'm bad at predicting future bugs. If you're using types it should be impossible to introduce that particular problem:

```python
    from enum import Enum, auto

    class Status(Enum):
        SUCCESS = auto()
        FAILURE = auto()

    status: Status
    status = Status.SUCCESS # this is fine
    status = "fake" # mypy complains here
```

Reply:
Oh I see, I guess you might forget to handle the new case, but I'd really expect that to be caught by code review.